src/main/java/com/yahoo/bullet/aggregations/Raw.java

```diff
@@ -0,0 +1,139 @@
+/*
+ *  Copyright 2017, Yahoo Inc.
+ *  Licensed under the terms of the Apache License, Version 2.0.
+ *  See the LICENSE file associated with the project for terms.
+ */
+package com.yahoo.bullet.aggregations;
+
+import com.yahoo.bullet.common.BulletConfig;
+import com.yahoo.bullet.common.BulletError;
+import com.yahoo.bullet.common.SerializerDeserializer;
+import com.yahoo.bullet.parsing.Aggregation;
+import com.yahoo.bullet.record.BulletRecord;
+import com.yahoo.bullet.result.Clip;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Implements the LIMIT operation on multiple raw {@link BulletRecord}.
+ *
+ * A call to {@link #getData()} or {@link #getResult()} will return the current collection of
+ * records, which is a {@link List} of {@link BulletRecord}.
+ *
+ * A call to {@link #combine(byte[])} with the result of {@link #getData()} will combine records from
+ * the {@link List} till the aggregation size is reached.
+ *
+ * This {@link Strategy} will only consume or combine till the specified aggregation size is reached.
+ */
+@Slf4j
+public class Raw implements Strategy {
+    private ArrayList<BulletRecord> aggregate = new ArrayList<>();
+
+    private Integer size;
+    private int consumed = 0;
+    private int combined = 0;
+
+    /**
+     * Constructor that takes in an {@link Aggregation} and a {@link BulletConfig}. The size of the aggregation is used
+     * as a LIMIT operation.
+     *
+     * @param aggregation The aggregation that specifies how and what this will compute.
+     * @param config The config that has relevant configs for this strategy.
+     */
+    @SuppressWarnings("unchecked")
+    public Raw(Aggregation aggregation, BulletConfig config) {
+        int maximumSize = config.getAs(BulletConfig.RAW_AGGREGATION_MAX_SIZE, Integer.class);
+        size = Math.min(aggregation.getSize(), maximumSize);
+    }
+
+    @Override
+    public boolean isClosed() {
+        return consumed + combined >= size;
+    }
+
+    @Override
+    public void consume(BulletRecord data) {
+        if (isClosed() || data == null) {
```

Comment:
For consume and combine, RAW checks isClosed, but others don't. Is it because no sketch in RAW? It's kinda not consistent with other queries.

Reply:
Yup, I can remove it. You're right - the inconsistency is weird. Since we made the change to make Querier not do any checking, we should remove it. It was just there as a safety to make sure users, if not honoring isClosed (or isClosedForPartition), don't consume and cause OOM issues.